code/sorting/quick_sort/quicksort.rs

```diff
@@ -0,0 +1,118 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+ 
+// This algorithm taken from https://github.com/servo/rust-quicksort
+
+//! In-place sorting.
+#![cfg_attr(not(test), no_std)]
+#[cfg(test)]
+extern crate std as core;
+
+use core::cmp::Ordering;
+
+fn quicksort_helper<T, F>(arr: &mut [T], left: isize, right: isize, compare: &F)
+where F: Fn(&T, &T) -> Ordering {
+    if right <= left {
+        return
+    }
+
+    let mut i: isize = left - 1;
+    let mut j: isize = right;
+    let mut p: isize = i;
+    let mut q: isize = j;
+    unsafe {
+        let v: *mut T = &mut arr[right as usize];
+        loop {
+            i += 1;
+            while compare(&arr[i as usize], &*v) == Ordering::Less {
+                i += 1
+            }
+            j -= 1;
+            while compare(&*v, &arr[j as usize]) == Ordering::Less {
+                if j == left {
+                    break
+                }
+                j -= 1;
+            }
+            if i >= j {
+                break
+            }
+            arr.swap(i as usize, j as usize);
+            if compare(&arr[i as usize], &*v) == Ordering::Equal {
+                p += 1;
+                arr.swap(p as usize, i as usize)
+            }
+            if compare(&*v, &arr[j as usize]) == Ordering::Equal {
+                q -= 1;
+                arr.swap(j as usize, q as usize)
+            }
+        }
+    }
+
+    arr.swap(i as usize, right as usize);
```

Comment:
Improve the indentation of the code .....this is not upto the coding standards ! 👍

Reply:
This code is using at Servo i think their style is like that i don't get the problem at this 👍